CMakeLists.txt

```diff
@@ -40,17 +58,40 @@ set(NOVELRT_VULKAN_VERSION "1.3.211" CACHE STRING "Vulkan version")
 set(NOVELRT_VULKAN_MEMORY_ALLOCATOR_VERSION "2.3.0" CACHE STRING "VulkanMemoryAllocator version")
 set(NOVELRT_VULKAN_SDK_PATH "" CACHE STRING "Directory for NovelRT to find the Vulkan SDK (mainly for macOS support)")
 
-find_package(Doxygen ${NOVELRT_DOXYGEN_VERSION}
-  COMPONENTS dot
-)
+if(NOVELRT_PIE)
+  set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+endif()
 
-if(NOT DOXYGEN_FOUND)
-  message("Doxygen ${NOVELRT_DOXYGEN_VERSION} was not found, building documentation is disabled.")
-  set(NOVELRT_BUILD_DOCUMENTATION OFF)
+# Remove unused symbols in non-debug mode.
+# This is useful even with LTO (-84 KiB with MinSizeRel).
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  # For some reason, adding to CMAKE_CXX_FLAGS results in a slightly smaller
+  # binary than using `add_compile/link_options`
+
+  set(_extra_flags "-ffunction-sections -fdata-sections")
+  if(APPLE)
+    set(_extra_flags "${_extra_flags} -Wl,-dead_strip")
+  else()
+    set(_extra_flags "${_extra_flags} -Wl,--gc-sections,--as-needed")
+  endif()
+
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${_extra_flags}")
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ${_extra_flags}")
+  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${_extra_flags}")
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} ${_extra_flags}")
+  set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} ${_extra_flags}")
+  set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_MINSIZEREL} ${_extra_flags}")
 endif()
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
These should use per-target properties:
https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html
https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html
https://cmake.org/cmake/help/latest/prop_tgt/EXPORT_COMPILE_COMMANDS.html